tests/rptest/transactions/verifiers/consumer_offsets_verifier.py

```diff
@@ -0,0 +1,195 @@
+        for producer in range(num_producers):
+            self._committed_offsets[f"group-{producer}"] = [
+                TopicPartition(self._source_topic, p, -1)
+                for p in range(num_src_partitions)
+            ]
+
+        self._total_commits_so_far = 0
+        self._max_commits = max_commits
+        self._commits_done = threading.Event()
+        threading.Thread(target=self._start_producers, daemon=True).start()
+
+    def _start_producers(self):
+        with ThreadPoolExecutor(max_workers=self._num_producers) as executor:
+            for producer in range(self._num_producers):
+                self._tasks.append(
+                    executor.submit(lambda: self._start_one_producer(
+                        group_id=f"group-{producer}", tx_id=f"txid-{producer}")
+                                    ))
+
+    def _stop_all(self, timeout_sec: int = 30):
+        if self._stop_ev.isSet():
+            return
+        self._stop_ev.set()
+        futures.wait(self._tasks,
+                     timeout=timeout_sec,
+                     return_when=futures.ALL_COMPLETED)
+
+    def _current_committed_offsets(self, group_id: str, partitions: list[int]):
+        with self._lock:
+            return [
+                tp for tp in self._committed_offsets[group_id]
+                if tp.partition in partitions
+            ]
+
+    def _update_committed_offsets(self, group_id: str,
+                                  positions: list[TopicPartition]):
+        with self._lock:
+            for position in positions:
+                self._committed_offsets[group_id][
+                    position.partition] = position
+            self._total_commits_so_far += 1
+            if self._total_commits_so_far >= self._max_commits:
+                self._commits_done.set()
+
+    def _start_one_producer(self, group_id: str, tx_id: str):
+
+        consumer = ck.Consumer({
+            'bootstrap.servers': self._redpanda.brokers(),
+            'group.id': group_id,
+            'auto.offset.reset': 'earliest',
+            'enable.auto.commit': False,
+        })
+
+        producer = ck.Producer({
+            'bootstrap.servers': self._redpanda.brokers(),
+            'transactional.id': tx_id,
+            'transaction.timeout.ms': 900000
+        })
+        producer.init_transactions()
+
+        def generate_dummy_positions():
+            # pick a random list of partitions to update
+            partitions = random.sample(range(0, self._num_src_partitions), 3)
+            current_offsets = self._current_committed_offsets(
+                group_id=group_id, partitions=partitions)
+            # update positions
+            for tp in current_offsets:
+                tp.offset = tp.offset + random.randint(1, 5)
+            return current_offsets
+
+        i = 0
+        while not self._stop_ev.isSet():
+            producer.begin_transaction()
+            producer.produce(self._topic, f"{tx_id}_{i}", f"{tx_id}_id")
```

Comment:
nit: is it true that in a real application we would actually read from the "source topic" and here we are just simulating it by randomly advancing offsets? IMO this deserves a comment, also maybe move "producing" to happen after "consuming"

Reply:
yes exactly, I think the for purpose of this test a real consumer is just noise, hence generating dummy offsets for validation. will update the comment.